plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/CreateMavenBasedAppEngineStandardProject.java

```diff
@@ -64,8 +69,15 @@ protected void execute(IProgressMonitor monitor)
         ? groupId : this.packageName;
     List<IProject> archetypeProjects = projectConfigurationManager.createArchetypeProjects(location,
         getArchetypeDescriptor(), groupId, artifactId, version, packageName, properties,
-        importConfiguration, progress.newChild(60));
+        importConfiguration, progress.newChild(40));
 
+    for (IProject project : archetypeProjects) {
+      IFacetedProject facetedProject = ProjectFacetsManager.create(
+          project, true, progress.newChild(10));
+      CreateAppEngineStandardWtpProject.installAppEngineFacet(facetedProject, monitor);  
+      CreateAppEngineStandardWtpProject.installAppEngineRuntime(facetedProject, monitor);  
```

Comment:
These calls should not be using `monitor` but should be splitting of work from the `progress`